specification/datalake-analytics/data-plane/Microsoft.DataLakeAnalytics/2017-09-01-preview/job.json

```diff
@@ -0,0 +1,1974 @@
+          "readOnly": true,
+          "format": "date-time",
+          "description": "the completion time of the job."
+        },
+        "state": {
+          "type": "string",
+          "readOnly": true,
+          "description": "the job state. When the job is in the Ended state, refer to Result and ErrorMessage for details.",
+          "enum": [
+            "Accepted",
+            "Compiling",
+            "Ended",
+            "New",
+            "Queued",
+            "Running",
+            "Scheduling",
+            "Starting",
+            "Paused",
+            "WaitingForCapacity"
+          ],
+          "x-ms-enum": {
+            "name": "JobState",
+            "modelAsString": false
+          }
+        },
+        "result": {
+          "type": "string",
+          "readOnly": true,
+          "description": "the result of job execution or the current result of the running job.",
+          "enum": [
+            "None",
+            "Succeeded",
+            "Cancelled",
+            "Failed"
+          ],
+          "x-ms-enum": {
+            "name": "JobResult",
+            "modelAsString": false
+          }
+        },
+        "logFolder": {
+          "type": "string",
+          "readOnly": true,
+          "description": "the log folder path to use in the following format: adl://<accountName>.azuredatalakestore.net/system/jobservice/jobs/Usql/2016/03/13/17/18/5fe51957-93bc-4de0-8ddc-c5a4753b068b/logs/."
+        },
+        "logFilePatterns": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "the list of log file name patterns to find in the logFolder. '*' is the only matching character allowed. Example format: jobExecution*.log or *mylog*.txt"
+        },
+        "related": {
+          "$ref": "#/definitions/JobRelationshipProperties",
+          "description": "the recurring job relationship information properties."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "the key-value pairs used to add additional metadata to the job information. (Only for use internally with Scope job type.)"
+        }
+      },
+      "required": [
+        "name",
+        "type"
+      ],
+      "description": "The common Data Lake Analytics job information properties."
+    },
+    "JobInformation": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/JobInformationBasic"
+        }
+      ],
+      "properties": {
+        "errorMessage": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/JobErrorDetails"
+          },
+          "description": "the error message details for the job, if the job failed."
+        },
+        "stateAuditRecords": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref"
```

Comment:
Consider using `x-ms-flatten`. Unless it'll introduce a breaking change.

Reply:
This is related to the "CreateJobProperties" topic above in that JobProperties is polymorphic via the discriminator value.  But JobInformation is primarily used for GETs now.  So, I would be in favor of flattening this.  However, it would be a breaking change. @matt1883, thoughts?

Reply:
Here is an example of the benefits: https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-client-flatten

Reply:
I would like to do this, but I confirmed with @olydis that not every language supports x-ms-flatten for property objects that are polymorphic via the discriminator value.  We will definitely consider this an upcoming release.  Thank you for mentioning this!